.github/workflows/upload.yml

```diff
@@ -0,0 +1,45 @@
+name: Upload to GCS
+
+on:
+  workflow_call:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  release:
+    if: github.repository == 'bufbuild/plugins'
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repository code
+        uses: actions/checkout@v3
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19.x
+          check-latest: true
+          cache: true
+      - name: Cache Plugins
+        id: cache-plugins
+        uses: actions/cache@v3
+        with:
+          path: downloads
+          key: plugin-downloads
```

Comment:
Caching this so we download less stuff each time. Might take this out if the cache uses too much storage, but for right now this should work.